File: backend/src/main/java/com/icms/model/Worker.java
package com.icms.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "workers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Worker {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String department;
    private String phone;
    private String email;

    @Enumerated(EnumType.STRING)
    private Status status;

    private LocalDateTime createdAt;

    public enum Status { AVAILABLE, BUSY, OFF_DUTY }

    @PrePersist
    void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) status = Status.AVAILABLE;
    }
}

File: backend/models/Worker.js
const mongoose = require('mongoose');

const workerSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    department: {
        type: String,
        trim: true
    },
    phone: {
        type: String,
        trim: true
    },
    email: {
        type: String,
        trim: true,
        lowercase: true
    },
    status: {
        type: String,
        enum: ['AVAILABLE', 'BUSY', 'OFF_DUTY'],
        default: 'AVAILABLE'
    }
}, {
    timestamps: { createdAt: 'createdAt', updatedAt: false }
});

const Worker = mongoose.model('Worker', workerSchema);
module.exports = Worker;
